### PR TITLE
Introduce StorageManager and refactor StorageService and StorageCommunicationEndpoint etc

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: shells/lib/components/arc-host.js [hunk 1]
```diff
@@ -15,12 +15,12 @@ import {devtoolsArcInspectorFactory} from '../../../build/devtools-connector/dev
 const {log, warn, error} = logsFactory('ArcHost', '#cade57');
 
 export class ArcHost {
-  constructor(context, storage, composer, storageservice, portFactories) {
+  constructor(context, storage, composer, storagemanager, portFactories) {
```
Comment (shans) [inline shells/lib/components/arc-host.js:18]: storagemanager should probably be camel case here (storageManager).

File: src/runtime/storage/storage-service.ts [hunk 2]
```diff
@@ -16,77 +16,10 @@ import {noAwait} from '../../utils/lib-utils.js';
 import {StoreInfo} from './store-info.js';
 import {StorageKey} from './storage-key.js';
 import {Exists} from './drivers/driver.js';
-
-export type StorageServiceCallback = (data: {}) => void;
+import {StorageEndpointManager} from './storage-manager.js';
+import {Consumer} from '../../utils/lib-utils.js';
 
 export interface StorageService {
```
Comment (shans) [inline src/runtime/storage/storage-service.ts:22]: Add a comment explaining what the StorageService is.

File: src/runtime/storage/storage-manager.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {CRDTTypeRecord} from '../../crdt//lib-crdt.js';
+import {TypeToCRDTTypeRecord} from './storage.js';
+import {ActiveStore} from './active-store.js';
+import {Type} from '../../types/lib-types.js';
+import {StoreInfo} from './store-info.js';
+import {StorageService} from './storage-service.js';
+
+export interface StorageEndpointManager {
```
Comment (shans) [inline src/runtime/storage/storage-manager.ts:17]: At a comment explaining what a StorageEndpointManager is.

**Review by shans — COMMENTED** [review]
This might be an issue from Kotlin but it seems like the name StorageEndpointManager and StorageService are backwards.

I'd expect the toplevel thing you grabbed to be the StorageService - it's the service for storage. If you need to do endpointy stuff I'd accept that there should be an EndpointManager inside it.

**Review by shans — APPROVED** [review]

File: src/runtime/storage/storage-service.ts (hunk 2 above)
Comment (mmandlis) [reply src/runtime/storage/storage-service.ts:25]: done

File: src/runtime/storage/storage-manager.ts (hunk 3 above)
Comment (mmandlis) [reply src/runtime/storage/storage-manager.ts:22]: done (and added TODOs for the next refactoring steps)

File: shells/lib/components/arc-host.js (hunk 1 above)
Comment (mmandlis) [reply shells/lib/components/arc-host.js]: done. 
(but i didn't completely make it up, in arc-component.js the observed attributes actually do not support camel case, must all be lowercase).

**Comment by mmandlis** [comment]
> This might be an issue from Kotlin but it seems like the name StorageEndpointManager and StorageService are backwards.
> 
> I'd expect the toplevel thing you grabbed to be the StorageService - it's the service for storage. If you need to do endpointy stuff I'd accept that there should be an EndpointManager inside it.

As discussed online, this change is consistent with how things are structured in KT. Might reconsider, but merging as is for now.

Next steps as agreed would be to refactor StorageCommunicationEndpointProvider be implemented by DirectStorageEndpointManager (instead of ActiveStore), and return StorageCommunicationEndpoints.

**Pull request closed without merging**